src/solace_agent_mesh/gateway/http_sse/repository/project_repository.py

```diff
@@ -0,0 +1,172 @@
+            ProjectModel.id == ProjectUserModel.project_id
+        ).filter(
+            ProjectModel.deleted_at.is_(None),  # Exclude soft-deleted projects
+            or_(
+                ProjectModel.user_id == user_id,
+                ProjectUserModel.user_id == user_id
+            )
+        ).distinct().all()
+        
+        return [self._model_to_entity(model) for model in models]
+
+    def get_filtered_projects(self, project_filter: ProjectFilter) -> List[Project]:
+        """Get projects based on filter criteria."""
+        query = self.db.query(ProjectModel).filter(
+            ProjectModel.deleted_at.is_(None)  # Exclude soft-deleted projects
+        )
+        
+        if project_filter.user_id is not None:
+            query = query.filter(ProjectModel.user_id == project_filter.user_id)
+
+        models = query.all()
+        return [self._model_to_entity(model) for model in models]
+
+    def get_by_id(self, project_id: str, user_id: str) -> Optional[Project]:
+        """
+        Get a project by its ID, ensuring user access.
+        
+        This checks if the user is the owner OR has access via project_users table.
+        """
+        model = self.db.query(ProjectModel).outerjoin(
+            ProjectUserModel,
+            ProjectModel.id == ProjectUserModel.project_id
+        ).filter(
+            ProjectModel.id == project_id,
+            ProjectModel.deleted_at.is_(None),  # Exclude soft-deleted projects
+            or_(
+                ProjectModel.user_id == user_id,
+                ProjectUserModel.user_id == user_id
+            )
+        ).first()
+        
+        return self._model_to_entity(model) if model else None
+
+    def update(self, project_id: str, user_id: str, update_data: dict) -> Optional[Project]:
+        """Update a project with the given data, ensuring user access."""
+        model = self.db.query(ProjectModel).filter(
+            ProjectModel.id == project_id,
+            ProjectModel.user_id == user_id,  # Only allow updates to user's own projects
+            ProjectModel.deleted_at.is_(None)  # Exclude soft-deleted projects
+        ).first()
+        
+        if not model:
+            return None
+        
+        for field, value in update_data.items():
+            if hasattr(model, field):
+                setattr(model, field, value)
+        
+        model.updated_at = now_epoch_ms()
+        self.db.commit()
+        self.db.refresh(model)
+        return self._model_to_entity(model)
+
+    def delete(self, project_id: str, user_id: str) -> bool:
+        """Delete a project by its ID, ensuring user access."""
+        result = self.db.query(ProjectModel).filter(
+            ProjectModel.id == project_id,
+            ProjectModel.user_id == user_id  # Only allow deletion of user's own projects
+        ).delete()
+        self.db.commit()
+        return result > 0
+
+    def soft_delete(self, project_id: str, user_id: str) -> bool:
```

Comment:
Should we also be soft-deleting the sessions associated with the project? Probably not, but worth a small discussion